cpp/src/hash/md5_hash.cu

```diff
@@ -301,7 +301,8 @@ std::unique_ptr<column> md5(table_view const& input,
                              }
                              return md5_leaf_type_check(col.type());
                            }),
-               "Unsupported column type for hash function.");
+               "Unsupported column type for hash function.",
+               cudf::data_type_error);
```

Comment:
The error type changes are causing some test failures in C++. Example:
```
 Expected: cudf::hashing::sha224(input) throws an exception of type cudf::logic_error.
  Actual: it throws cudf::data_type_error with description "CUDF failure at: /opt/conda/conda-bld/work/cpp/src/hash/sha_hash.cuh:512: Unsupported column type for hash function.".
```

Please update the tests accordingly.